### PR TITLE
First and Last days extensions added

### LINKED ISSUE #50: [Feature Request] New methods on DateTime (firstDayOfMonth, lastDayOfMonth, lastDayOfWeek, firstDayOfWeek, firstDayOfYear, lastDayOfYear)
Looking for `lastDayOfMonth` like feature, found [this StackOverflow question](https://stackoverflow.com/questions/14814941/how-to-find-last-day-of-month) and I really liked the way [Yogesh Parwani](https://stackoverflow.com/users/9988524/yogesh-parwani) mixed [Chris Buckett](https://stackoverflow.com/users/444738/chris-buckett) and [Kai Sellgren](https://stackoverflow.com/users/1364206/kai-sellgren) awnsers. As pointed by [Juniper Belmont](https://stackoverflow.com/users/1724321/juniper-belmont) on Buckett's answer, I do think this could be a new feature as [mnordine](https://github.com/mnordine) suggested [here](https://github.com/dart-lang/sdk/issues/48943).

(Better code from [here](https://github.com/dart-lang/sdk/issues/48943#issuecomment-1115903432) - [lrhn](https://github.com/lrhn))

Code bellow:
```dart
extension DateTimeFirstLast on DateTime {
  DateTime get firstDayOfWeek => DateTime.utc(year, month, day + 1 - weekday);
  DateTime get lastDayOfWeek => DateTime.utc(year, month, day + 7 - weekday);  
  DateTime get firstDayOfMonth => DateTime.utc(year, month, 1);
  DateTime get lastDayOfMonth => DateTime.utc(year, month + 1, 0);
  DateTime get firstDayOfYear => DateTime.utc(year, 1, 1);
  DateTime get lastDayOfYear => DateTime.utc(year, 12, 31);
}
```

### PR DESCRIPTION
**Pull request by FMorschel**
Fixes #50.

### CODE REVIEW

File: lib/src/extensions.dart [hunk 1]
```diff
@@ -204,6 +204,24 @@ extension DateTimeTimeExtension on DateTime {
             microsecond ?? this.microsecond,
           );
   }
+
+  /// Returns the Monday of this week
+  DateTime get firstDayOfWeek => DateTime.utc(year, month, day + 1 - weekday);
```
Comment (jogboms) [inline lib/src/extensions.dart:209]: I would like us to check if the current `DateTime` `isUtc` before constructing a `utc` variant. 
For example, see line 185

**Comment by jogboms** [comment]
I approve of this @FMorschel but do you plan to add some test cases for these?

**Comment by jogboms** [comment]
For the formatting, you can set your ide line width for this project to 120. @FMorschel

**Comment by FMorschel** [comment]
> I approve of this @FMorschel but do you plan to add some test cases for these?

Tests created

**Review by jogboms — APPROVED** [review]
lgtm-f1affd

**Pull request merged**